airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -731,24 +733,42 @@ def __init__(self, replication_key: str, stream_slice_step: str = "P30D", **kwar
         super().__init__(**kwargs)
         self.replication_key = replication_key
         self._stream_slice_step = stream_slice_step
+        self._cursor = None
+
+    @property
+    def cursor(self) -> Optional[Cursor]:
+        if not self._cursor:
+            raise ValueError("Cursor should be set at this point")
+        return self._cursor
+
+    @cursor.setter
+    def cursor(self, cursor: Cursor):
+        self._cursor = cursor
 
     def stream_slices(
         self, *, sync_mode: SyncMode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
     ) -> Iterable[Optional[Mapping[str, Any]]]:
-        now = pendulum.now(tz="UTC")
-        assert LOOKBACK_SECONDS is not None and LOOKBACK_SECONDS >= 0
-
-        initial_date = self.get_start_date_from_state(stream_state) - pendulum.Duration(seconds=LOOKBACK_SECONDS)
-        slice_start = initial_date
-        while slice_start < now:
-            slice_end = slice_start + self.stream_slice_step
-            self._slice = {
-                "start_date": slice_start.isoformat(timespec="milliseconds"),
-                "end_date": min(slice_end, now).isoformat(timespec="milliseconds"),
-            }
-            yield self._slice
-
-            slice_start += self.stream_slice_step
+        if self._cursor:
```

Comment:
I really hate this but for full refresh, we create a `FinalStateCursor` and not a `ConcurrentCursor`. It probably means that the generation of slices based on a state is not valid. In other words, with or without a state, we generate slices that relies on incrementality and the fact that the stream is full_refresh is irrelevant. I think this will be more true with Resumable Full Refresh but it is a bit annoying right now